objects/src/accounts/storage/map.rs

```diff
@@ -85,12 +98,46 @@ impl StorageMap {
         self.map.inner_nodes() // Delegate to Smt's inner_nodes method
     }
 
-    // STATE MUTATORS
+    // DATA MUTATORS
     // --------------------------------------------------------------------------------------------
-
     pub fn insert(&mut self, key: RpoDigest, value: Word) -> Word {
         self.map.insert(key, value) // Delegate to Smt's insert method
     }
+
+    /// Applies the provided delta to this account storage.
+    ///
+    /// This method assumes that the delta has been validated by the calling method and so, no
+    /// additional validation of delta is performed.
+    pub fn apply_delta(&mut self, delta: &StorageMapDelta) -> Result<(), AccountError> {
+        // apply the updated leaves to the storage map
+        for &(key, value) in delta.updated_leaves.iter() {
+            self.set_map_item(key, value)?;
+        }
+
+        // apply the cleared leaves to the storage map
+        // currently we cannot remove leaves from the storage map, so we just set them to empty
+        for &key in delta.cleared_leaves.iter() {
+            self.set_map_item(key, EMPTY_WORD)?;
+        }
+
+        Ok(())
```

Comment:
Not for this PR - but this should probably return the new root of the storage map.